File: mobile/src/main/java/at/shockbytes/remote/network/model/SlidesResponse.java
package at.shockbytes.remote.network.model;

import java.util.List;

/**
 * @author Martin Macheiner
 *         Date: 15.10.2017.
 */

public class SlidesResponse {

    private final String name;
    private final int errorCode;
    private final List<SlidesEntry> slides;
    private final int slideAmount;

    public SlidesResponse(String name, int errorCode, List<SlidesEntry> slides, int slideAmount) {
        this.name = name;
        this.errorCode = errorCode;
        this.slides = slides;
        this.slideAmount = slideAmount;
    }

    public String getName() {
        return name;
    }

    public int getErrorCode() {
        return errorCode;
    }

    public int getSlideAmount() {
        return slideAmount;
    }

    public List<SlidesEntry> getSlides() {
        return slides;
    }

    @Override
    public String toString() {

        StringBuilder sb = new StringBuilder();
        sb.append("Name: ").append(getName()).append("\n");
        sb.append("Error code: ").append(getErrorCode()).append("\n");
        sb.append("Slide amount: ").append(getSlideAmount()).append("\n");
        for (SlidesEntry e : slides) {
            sb.append(e.toString());
        }
        return sb.toString();
    }

    public class SlidesEntry {

        private final byte[] base64Image;
        private final String notes;
        private final int slideNumber;

        public SlidesEntry(byte[] base64Image, String notes, int slideNumber) {
            this.base64Image = base64Image;
            this.notes = notes;
            this.slideNumber = slideNumber;
        }

        public byte[] getBase64Image() {
            return base64Image;
        }

        public String getNotes() {
            return notes;
        }

        public int getSlideNumber() {
            return slideNumber;
        }


        @Override
        public String toString() {
            return "\tBase64 length: " + (base64Image != null ? base64Image.length : -1)
                    + "\n\tNotes: " + notes
                    + "\n\tSlide number: " + slideNumber + "\n-------------\n";
        }
    }

}

File: mobile/src/main/java/at/shockbytes/remote/network/model/SlidesResponse.kt
package at.shockbytes.remote.network.model

import java.util.*

/**
 * Author:  Mescht
 * Date:    16.10.2017
 */

data class SlidesResponse(val name : String = "", val errorCode : Int,
                          val slides : List<SlidesEntry>, val slideAmount : Int) {

    data class SlidesEntry(val base64Image: ByteArray = ByteArray(0),
                           val notes: String = "", val slideNumber : Int) {

        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false

            other as SlidesEntry

            if (!Arrays.equals(base64Image, other.base64Image)) return false
            if (notes != other.notes) return false
            if (slideNumber != other.slideNumber) return false

            return true
        }

        override fun hashCode(): Int {
            var result = Arrays.hashCode(base64Image)
            result = 31 * result + notes.hashCode()
            result = 31 * result + slideNumber
            return result
        }
    }
}